wp-modules/app/js/src/components/PatternPreview/test/PreviewIframe.spec.tsx

```diff
@@ -0,0 +1,114 @@
+/* eslint-disable import/default, camelcase */
+
+import { create } from 'react-test-renderer';
+import PreviewIframe from '../PreviewIframe';
+import WP_Element from '@wordpress/element';
+
+const FAKE_URL = 'http://some-fake-url.com';
+const VIEWPORT_WIDTH = 1200;
+const SCALE = 0.5;
+const STATE_MOCK_$iframeRef = {
+	contentWindow: {
+		document: {
+			documentElement: {
+				scrollHeight: 1200,
+			},
+		},
+	},
+};
+const STATE_MOCK_$iframeBodyHeight = 600;
+const STATE_MOCK_$iframeLoaded = true;
+
+jest.mock( '@wordpress/element', () => {
+	return {
+		...jest.requireActual( '@wordpress/element' ),
+		useState: jest.fn(),
+	};
+} );
+
+describe( 'PreviewIframe', () => {
+	beforeEach( () => {
+		jest.spyOn( WP_Element, 'useState' )
+			.mockImplementationOnce( () => [
+				STATE_MOCK_$iframeRef,
+				() => null,
+			] )
+			.mockImplementationOnce( () => [
+				STATE_MOCK_$iframeBodyHeight,
+				() => null,
+			] )
+			.mockImplementationOnce( () => [
+				STATE_MOCK_$iframeLoaded,
+				() => null,
+			] );
+	} );
```

Comment:
Maybe a bit hacky, but this is an option for mocking state values in the component.

Reply:
I know, I wish there was an easier way to do the same kind of thing that wasn't so easy to break.

Reply:
Looking at `PreviewIframe` again for creating these tests, I do wonder if we could get rid of at least a few of those `useState` calls. But that's for another PR!

Reply:
Good idea